File: cpp/template.cpp
#include <bits/stdc++.h>
using namespace std;

auto solve(/* params */) {
    /*
    SPEC:
    - Entradas:
    - Salidas:
    - Reglas/empates:
    - Edge cases:
    */
    // TODO: implementar
    return /* tipo */;
}

struct TestCase {
    tuple</* tipos entrada */> inp;
    /* tipo salida */ expected;
};

int main() {
    vector<TestCase> cases = {
        { make_tuple(/* ... */), /* expected */ },
        { make_tuple(/* ... */), /* expected */ },
        { make_tuple(/* ... */), /* expected */ },
    };

    int fails = 0;
    for (size_t i = 0; i < cases.size(); ++i) {
        auto [/* arg1, arg2, ... */] = cases[i].inp;
        auto got = solve(/* arg1, arg2, ... */);
        if (!(got == cases[i].expected)) {
            cerr << "FAIL case " << i+1 << "\n";
            fails++;
        }
    }
    if (fails) {
        cerr << fails << " test(s) failed\n";
        return 1;
    }
    cout << "OK\n";
}

File: rust/template.rs
pub fn solve(/* params */) -> /* tipo salida */ {
    /*
    SPEC:
    - Entradas:
    - Salidas:
    - Reglas/empates:
    - Edge cases:
    */
    // TODO: implementar
    unimplemented!()
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn table() {
        let cases = vec![
            ((/* arg1 */, /* arg2 */), /* expected */),
            ((/* ... */), /* expected */),
            ((/* ... */), /* expected */),
        ];

        for (i, (inp, expected)) in cases.into_iter().enumerate() {
            let (/* arg1, arg2, ... */) = inp;
            let got = solve(/* arg1, arg2, ... */);
            assert_eq!(got, expected, "case {}", i+1);
        }
    }
}
